### ISSUE TITLE
Feature Request: Always Focus Main Window When Cycling Screens

### ISSUE THREAD

**Issue by wadegbow** [ianyh/Amethyst#1623]
**Is your feature request related to a problem? Please describe.**
When I cycle screens, a majority of the time the logic in `FocusTransitionCoordinator focusScreen()` seems to fail and my last focused window is not focused. Maybe I'm misunderstanding the logic.

**Describe the solution you'd like**
It would be nice if there was a setting to automatically select the main window when you cycle screens. 95% of the time I cycle screens I am wanting to focus the main window.

**Additional context**
- the last focused window is not properly being focused when cycling screens, maybe this is a bug?
- regardless of this is a bug I still think that always focusing the main window when cycling screens would be a useful option
- cycling spaces seems to always select the last focused window unlike cycling screens

**Comment by adisidev** [comment]
Agreed! To clarify, the most intuitive thing to happen would be when I cycle to a particular screen, whatever window I can see at the moment is the one that gets focused. I do actually think this is a bug, since I find my last focused window also not getting focused. @wadegbow I'd mark this as a bug if I were you.

@ianyh thank you so much for all your work with amethyst! I'd imagine this would be a bug that a lot of people face but haven't verbalised here. If there's any way I can help fixing this, please let me know, and I'll try my level best to follow any high level instructions you give.
